main.py

```diff
@@ -375,43 +375,80 @@ def update_preferences():
     if cur_user is None:
         return jsonify(respo='user not found')
     cur_report_preferences = db_session.query(ReportPreferences).filter(User.id == cur_id).first()
+    cur_general_preferences = db_session.query(GeneralPreferences).filter(User.id == cur_id).first()
     if request.method == "GET":
-        if cur_report_preferences is None:
-            return jsonify(lat='', lon='', distance='', accident_severity='0', history_report=False)
+        if cur_report_preferences is None and cur_general_preferences is None:
+            return jsonify(accident_severity='0', prefAccidentsLms=True, prefAccidentsIhud=True, produceAccidentsReprot=False)
         else:
-            return jsonify(lat=cur_report_preferences.latitude, lon=cur_report_preferences.longitude, distance=cur_report_preferences.radius, \
-                accident_severity=cur_report_preferences.minimum_severity, history_report=cur_report_preferences.historical_report)
+            resource_type = cur_general_preferences.resource_type.split(',')
+            prefAccidentsLms = False
+            prefAccidentsIhud = False
+            for word in resource_type:
+                if word == 'Lms':
+                    prefAccidentsLms = True
+                elif word == 'Ihud':
+                    prefAccidentsIhud = True
+            if cur_report_preferences is None:
```

Comment:
1. Use underscores for all variable names, not camel case.
2. "reprot" should be "report".
3. Why not use a list of data sources to include, rather than adding a new boolean variable for each data source? Then you can use the numeric codes for the sources.